### PR TITLE
Extract Button component

### PR DESCRIPTION
**Pull request by rudolfs**
We avoid using brittle global styling by extracting buttons into a separate component.

Screenshot 2022-09-05 at 15 38 35

### CODE REVIEW

File: src/Connect.svelte [hunk 1]
```diff
@@ -34,18 +34,17 @@
...
 
-<button
-  on:click|stopPropagation={onConnect}
-  {style}
-  class="connect {className}"
+<Button
+  on:click={onConnect}
+  variant={buttonVariant}
   disabled={connecting}
   data-waiting={connecting || null}>
```
Comment (cloudhead) [inline src/Connect.svelte:41]: Data properties won't make it through as-is

Comment (rudolfs) [reply src/Connect.svelte]: Good catch! 4ebbf94d3205f573f786863e9a10c682c38a04d0

**Comment by cloudhead** [comment]
I think this is good-to-go, code-wise, just need to make sure there's no design regressions since a lot of styles moved around!

**Comment by cloudhead** [comment]
Here's one I'm guessing from this PR:
image

**Comment by rudolfs** [comment]
I actually compared most of the screens with production and they lined up.

Screenshot 2022-09-06 at 11 32 13


Screenshot 2022-09-06 at 11 33 04

The only screens I didn't test were Orgs, because we're removing them anyway in https://github.com/radicle-dev/radicle-interface/pull/359.

**Comment by cloudhead** [comment]
Ah damn, then this must come from some other change :thinking:

**Comment by rudolfs** [comment]
> Ah damn, then this must come from some other change 🤔

Very low on my todo list, but I wanted to set up Playwright at some point and make it do screenshots of some of the happy-paths, so that we can spot these regressions more easily with screenshot diffing.

**Pull request merged**